src/NovelAudioService.cpp

```diff
@@ -0,0 +1,367 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT License (MIT). See LICENCE.md in the repository root for more information.
+
+#include "NovelAudioService.h"
+#include <iostream>
+#include <functional>
+
+namespace NovelRT {
+
+NovelAudioService::NovelAudioService() : _nextChannel(1), _musicTime(0), _musicPausedTime(0), isInitialized(false) {
+  initializeAudio();
+}
+
+bool NovelAudioService::initializeAudio() {
+  if (SDL_InitSubSystem(SDL_INIT_AUDIO) < 0 )
+  {
+    std::cout << "ERROR: Cannot play audio!" << std::endl;
+    std::cout << "SDL Error: " << SDL_GetError() << std::endl;
+  }
+  else if (Mix_OpenAudio(44100, MIX_DEFAULT_FORMAT, 2, 2048) < 0)
+  {
+    std::cout << "ERROR: Cannot play audio!" << std::endl;
+    std::cout << "SDL_Mixer Error: " << Mix_GetError() << std::endl;
+  }
+  else if (Mix_AllocateChannels(2) < 0)
+  {
+    std::cout << "ERROR: Failed to allocate channels." << std::endl;
+  }
+  else
+  {
+    std::cout << "INFO: Mixer engaged!" << std::endl;
+    isInitialized = true;
+  }
+  return isInitialized;
+}
+
+void NovelAudioService::loadSound(std::string input) {
```

Comment:
Not for this PR, but as a future design note.

It would be good if you gave back some kind of "opaque handle" that could be used instead of the file name. Basically just something that, for an already loaded sound, can be used to play a sound with `O(1)` (or really cheap) lookup.

Reply:
Issue created!